pkg/controller/daemon/daemon_controller.go

```diff
@@ -865,12 +899,29 @@ func (dsc *DaemonSetsController) podsShouldBeOnNode(
 				continue
 			}
 			if pod.Status.Phase == v1.PodFailed {
+				failedPodsObserved++
+
+				// This is a critical place where DS is often fighting with kubelet that rejects pods.
+				// We need to avoid hot looping and backoff.
+				backoffKey := fmt.Sprintf("%s/%s", dsKey, node.Name)
+				delay := dsc.failedPodsBackoff.Get(backoffKey)
+
+				now := time.Now()
+				inBackoff := dsc.failedPodsBackoff.IsInBackOffSinceUpdate(backoffKey, now)
```

Comment:
nit: `isInBackoff`

Reply:
my rule of thumb here is the the function would be called `isInBackoff`, the variable is state so only `inBackoff` - not sure how native English speaker would see it